### PR TITLE
First round of fixes in making gossip publishing good for the validator: See comment

### PR DESCRIPTION
**Pull request by Giulio2002**
* Fixed and simplified unaggregated bits check.
   * There are 2 bits on, one for the attester and one for the End-of-bitlist, needed to account for end of bitlist bit
 * Wrong publishing topic for sync_committee_ messages
 * Added more Ignore by receiving specific errors to avoid forwarding useless data.
 * Replaced `validateAttestation` with full message processing
 * Fixed forwarding of sync committee aggregates
 * Fixed subnet announcements

### CODE REVIEW

File: cl/beacon/handler/pool.go [hunk 1]
```diff
@@ -64,9 +65,20 @@ func (a *ApiHandler) PostEthV1BeaconPoolAttestations(w http.ResponseWriter, r *h
...
 	failures := []poolingFailure{}
 	for i, attestation := range req {
-		if err := a.forkchoiceStore.ValidateOnAttestation(attestation); err != nil {
+		var (
+			slot                  = attestation.AttestantionData().Slot()
+			cIndex                = attestation.AttestantionData().CommitteeIndex()
+			committeeCountPerSlot = headState.CommitteeCount(slot / uint64(a.beaconChainCfg.SlotsPerEpoch))
+			subnet                = subnets.ComputeSubnetForAttestation(committeeCountPerSlot, slot, cIndex, a.beaconChainCfg.SlotsPerEpoch, a.netConfig.AttestationSubnetCount)
+		)
+		if err := a.attestationService.ProcessMessage(r.Context(), &subnet, attestation); err != nil {
```
Comment (kevin-stackfi) [inline cl/beacon/handler/pool.go:81]: can you help to remove `a.committeeSub.CheckAggregateAttestation` ? which is already included in `a.attestationService.ProcessMessage`

Comment (domiwei) [inline cl/beacon/handler/pool.go:81]: can you help to remove `a.committeeSub.CheckAggregateAttestation` ? which is already included in `a.attestationService.ProcessMessage`

Comment (Giulio2002) [reply cl/beacon/handler/pool.go:83]: yeah removed

**Comment by Giulio2002** [comment]
I think next steps are:
* Network search for subnet peers
* Some experiments to see if we forward bad blocks ourselves

Symptoms remained:
* Sync committee messages are erratically forwarded (not so often).
* Attestations seems to never make it to the network.
* Blocks publishing stops working eventually.

File: cl/phase1/network/services/sync_committee_messages_service.go [hunk 2]
```diff
@@ -52,6 +55,8 @@ func NewSyncCommitteeMessagesService(
 
 // ProcessMessage processes a sync committee message
 func (s *syncCommitteeMessagesService) ProcessMessage(ctx context.Context, subnet *uint64, msg *cltypes.SyncCommitteeMessage) error {
+	s.mu.Lock()
```
Comment (domiwei) [inline cl/phase1/network/services/sync_committee_messages_service.go:58]: do we need to protect the whole ProcessMessage here?

File: cl/sentinel/gossip.go [hunk 3]
```diff
@@ -123,7 +152,14 @@ func (s *GossipManager) Recv() <-chan *GossipMessage {
 func (s *GossipManager) GetMatchingSubscription(match string) *GossipSubscription {
 	var sub *GossipSubscription
 	s.subscriptions.Range(func(topic, value interface{}) bool {
-		if strings.Contains(topic.(string), match) {
+		topicStr := topic.(string)
+		// take out third part of the topic by splitting on "/"
+		// reference: eth2/d31f6191/beacon_attestation_45/ssz_snappy
```
Comment (domiwei) [inline cl/sentinel/gossip.go:157]: seems the example is wrong becasue miss the leading '/'
`/eth2/d31f6191/beacon_attestation_45/ssz_snappy`

File: cl/sentinel/utils.go [hunk 4]
```diff
@@ -121,52 +122,65 @@ func (s *Sentinel) inMeshCap() float64 {
...
 	}
-	s.listener.LocalNode().Set(enr.WithEntry(s.cfg.NetworkConfig.SyncCommsSubnetKey, subnetField))
+	s.listener.LocalNode().Set(enr.WithEntry(s.cfg.NetworkConfig.SyncCommsSubnetKey, &subnetField))
 }
 
 // updateENROnSubscription updates the ENR based on the subscription status to subnets/syncnets.
 func (s *Sentinel) updateENROnSubscription(topicName string, subscribe bool) {
 	s.metadataLock.Lock()
 	defer s.metadataLock.Unlock()
+	///eth2/d31f6191/beacon_attestation_45/ssz_snappy
```
Comment (domiwei) [inline cl/sentinel/utils.go:167]: wrong example here
```suggestion
	//. topic: /eth2/d31f6191/beacon_attestation_45/ssz_snappy
```

File: cl/sentinel/gossip.go [hunk 5]
```diff
@@ -292,6 +317,169 @@ func (s *Sentinel) defaultVoluntaryExitTopicParams() *pubsub.TopicScoreParams {
 	}
 }
 
+func (s *Sentinel) defaultSyncSubnetTopicParams(activeValidators uint64) *pubsub.TopicScoreParams {
```
Comment (domiwei) [inline cl/sentinel/gossip.go:320]: hmm...
i should read libp2p to get more understand on it, right?

**Review by domiwei — APPROVED** [review]

File: cl/phase1/network/services/sync_committee_messages_service.go (hunk 2 above)
Comment (Giulio2002) [reply cl/phase1/network/services/sync_committee_messages_service.go:58]: because we can call it concurrently

File: cl/sentinel/gossip.go (hunk 5 above)
Comment (Giulio2002) [reply cl/sentinel/gossip.go:320]: I copied it from Prysm. but if you want to learn it https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.0.md

**Pull request merged**